File: study-planner-frontend/src/api/ml.js
import axios from 'axios';

const API_BASE_URL = 'http://localhost:8000/api/ml';

export const trainModel = async () => {
  const response = await axios.post(`${API_BASE_URL}/train`);
  return response.data;
};

export const predictHours = async (pastScore, difficulty, chapters, daysLeft) => {
  const response = await axios.post(`${API_BASE_URL}/predict`, null, {
    params: {
      past_score: pastScore,
      difficulty: difficulty,
      chapters: chapters,
      days_left: daysLeft
    }
  });
  return response.data;
};

File: study-planner-backend/app/routers/ml.py
from fastapi import APIRouter
from app.services.ml_model import train_model, predict_hours

router = APIRouter(prefix="/api/ml", tags=["ml"])

@router.post("/train")
def train_ml_model():
    """Train the ML model"""
    result = train_model()
    return {"message": result}

@router.post("/predict")
def predict_study_hours(past_score: float, difficulty: str, chapters: int, days_left: int):
    """Predict recommended study hours"""
    hours = predict_hours(past_score, difficulty, chapters, days_left)
    return {
        "predicted_hours": hours,
        "inputs": {
            "past_score": past_score,
            "difficulty": difficulty,
            "chapters": chapters,
            "days_left": days_left
        }
    }

